mule-packager/src/main/java/org/mule/tools/api/packager/archiver/Archiver.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Mule ESB Maven Tools
+ * <p>
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * <p>
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.tools.api.packager.archiver;
+
+import org.apache.commons.lang3.StringUtils;
+import org.codehaus.plexus.archiver.ArchiverException;
+import org.codehaus.plexus.archiver.zip.ZipArchiver;
+import org.codehaus.plexus.util.DirectoryScanner;
+
+import java.io.File;
+
+import static org.mule.tools.api.packager.structure.FolderNames.MAVEN;
+import static org.mule.tools.api.packager.structure.FolderNames.META_INF;
+
+public class Archiver extends ZipArchiver {
```

Comment:
please make this abstract and rename it to AbstractArchiver also please add javadoc to the class